File: server/DAL/Models/Dessert.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DAL.Models
{
    public class Dessert
    {
        public int Id { get; set; }
        public string Name { get; set; } 
        public string ImageUrl { get; set; }
        public double Price { get; set; }

        public ICollection<DessertIngredient> Ingredients { get; set; }
    }
}

File: client/bakery-test/src/app/viewModels/dessert.ts
export interface Dessert {
    id: number;
    name: string;
    imageUrl: string;
    price: number;
    ingredients: DessertIngredient[];
}

export interface DessertIngredient {
    id: number;
    amount: number;
    unit: string;
    name: string;
}
